### PR TITLE
tidb_row_checksum(): Update docs

### PR DESCRIPTION
**Pull request by dveeden**
### What is changed, added or deleted? (Required)

1. The statement in the example had syntax errors
2. The variable docs didn't link to the function

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v8.2 (TiDB 8.2 versions)
- [x] v8.1 (TiDB 8.1 versions)
- [ ] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

- #13258 
- https://github.com/pingcap/tidb/pull/43479

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by dveeden** [comment]
This doesn't fix all issues:

For me the example returns `NULL` instead of the checksum:

```
sql> SELECT TIDB_VERSION()\G
*************************** 1. row ***************************
TIDB_VERSION(): Release Version: v8.1.0
Edition: Community
Git Commit Hash: 945d07c5d5c7a1ae212f6013adfb187f2de24b23
Git Branch: HEAD
UTC Build Time: 2024-05-21 03:51:57
GoVersion: go1.21.10
Race Enabled: false
Check Table Before Drop: false
Store: tikv
1 row in set (0.0011 sec)

sql> SET GLOBAL tidb_enable_row_level_checksum = ON;
Query OK, 0 rows affected (0.0223 sec)

sql> USE test;
Default schema set to `test`.
Fetching global names, object names from `test` for auto-completion... Press ^C to stop.

sql> CREATE TABLE t (id INT PRIMARY KEY, k INT, c CHAR(1));
Query OK, 0 rows affected (0.1103 sec)

sql> INSERT INTO t values (1, 10, 'a');
Query OK, 1 row affected (0.0045 sec)

sql> SELECT *, TIDB_ROW_CHECKSUM() FROM t WHERE id = 1;
+----+----+---+---------------------+
| id | k  | c | TIDB_ROW_CHECKSUM() |
+----+----+---+---------------------+
|  1 | 10 | a | NULL                |
+----+----+---+---------------------+
1 row in set (0.0014 sec)
```

2. I don't think the "FastPlan" that is mentioned here is documented well enough.

Is "FastPlan" a generic term for `PointGet` and `TableDual` and `Set`?

https://github.com/pingcap/tidb/blob/df64c343ddfc4063891900acf95e2b3640c79f29/pkg/executor/adapter.go#L398-L417

**Comment by dveeden** [comment]
Looks like a re-connect is needed for this to work:

```
dvaneeden@dve-carbon:~$ mysqlsh mysql://root@127.0.0.1:4000/test

sql> SHOW GLOBAL VARIABLES LIKE 'tidb_enable_row_level_checksum';
+--------------------------------+-------+
| Variable_name                  | Value |
+--------------------------------+-------+
| tidb_enable_row_level_checksum | OFF   |
+--------------------------------+-------+
1 row in set (0.0009 sec)

sql> SHOW SESSION VARIABLES LIKE 'tidb_enable_row_level_checksum';
+--------------------------------+-------+
| Variable_name                  | Value |
+--------------------------------+-------+
| tidb_enable_row_level_checksum | OFF   |
+--------------------------------+-------+
1 row in set (0.0013 sec)

sql> SET GLOBAL tidb_enable_row_level_checksum = ON;
Query OK, 0 rows affected (0.0186 sec)

sql> SHOW GLOBAL VARIABLES LIKE 'tidb_enable_row_level_checksum';
+--------------------------------+-------+
| Variable_name                  | Value |
+--------------------------------+-------+
| tidb_enable_row_level_checksum | ON    |
+--------------------------------+-------+
1 row in set (0.0014 sec)

sql> SHOW SESSION VARIABLES LIKE 'tidb_enable_row_level_checksum';
+--------------------------------+-------+
| Variable_name                  | Value |
+--------------------------------+-------+
| tidb_enable_row_level_checksum | ON    |
+--------------------------------+-------+
1 row in set (0.0015 sec)

sql> USE test;
Default schema set to `test`.
Fetching global names, object names from `test` for auto-completion... Press ^C to stop.

sql> CREATE TABLE t (id INT PRIMARY KEY, k INT, c CHAR(1));
Query OK, 0 rows affected (0.1069 sec)

sql> INSERT INTO t values (1, 10, 'a');
Query OK, 1 row affected (0.0124 sec)

sql> SELECT *, TIDB_ROW_CHECKSUM() FROM t WHERE id = 1;
+----+----+---+---------------------+
| id | k  | c | TIDB_ROW_CHECKSUM() |
+----+----+---+---------------------+
|  1 | 10 | a | NULL                |
+----+----+---+---------------------+
1 row in set (0.0016 sec)

sql> 
Bye!
dvaneeden@dve-carbon:~$ mysqlsh mysql://root@127.0.0.1:4000/test

sql> SELECT *, TIDB_ROW_CHECKSUM() FROM t WHERE id = 1;
+----+----+---+---------------------+
| id | k  | c | TIDB_ROW_CHECKSUM() |
+----+----+---+---------------------+
|  1 | 10 | a | NULL                |
+----+----+---+---------------------+
1 row in set (0.0016 sec)

sql> delete from t;
Query OK, 1 row affected (0.0038 sec)

sql> INSERT INTO t values (1, 10, 'a');
Query OK, 1 row affected (0.0112 sec)

sql> SELECT *, TIDB_ROW_CHECKSUM() FROM t WHERE id = 1;
+----+----+---+---------------------+
| id | k  | c | TIDB_ROW_CHECKSUM() |
+----+----+---+---------------------+
|  1 | 10 | a | 3813955661          |
+----+----+---+---------------------+
1 row in set (0.0018 sec)

sql> 
```

**Comment by dveeden** [comment]
I'm not sure if the intended use of this function is clearly described. Is this only for TiCDC? Or is this also useful in other cases? How does this relate to `ADMIN CHECKSUM TABLE`?

**Review by 3AceShowHand — APPROVED** [review]

File: system-variables.md [hunk 1]
```diff
@@ -2771,6 +2771,8 @@ Query OK, 0 rows affected (0.09 sec)
 
 </CustomContent>
 
+- The [`TIDB_ROW_CHECKSUM()`](/functions-and-operators/tidb-functions.md#tidb_row_checksum) function can be used to get the checksum of a row.
```
Comment (hfxsd) [inline system-variables.md:2774]:
```suggestion
- You can use the [`TIDB_ROW_CHECKSUM()`](/functions-and-operators/tidb-functions.md#tidb_row_checksum) function to get the checksum of a row.
```

**Review by hfxsd — APPROVED** [review]

File: system-variables.md [hunk 2]
```diff
@@ -2771,6 +2771,8 @@ Query OK, 0 rows affected (0.09 sec)
 
 </CustomContent>
 
+- You can use the [`TIDB_ROW_CHECKSUM()`](/functions-and-operators/tidb-functions.md#tidb_row_checksum) function to get the checksum of a row.
```
Comment (Oreoxmt) [inline system-variables.md:2774]:
```suggestion
- You can use the [`TIDB_ROW_CHECKSUM()`](/functions-and-operators/tidb-functions.md#tidb_row_checksum) function to get the checksum value of a row.
```

File: functions-and-operators/tidb-functions.md [hunk 3]
```diff
@@ -333,12 +333,14 @@ To enable the checksum feature of single-row data in TiDB (controlled by the sys
 SET GLOBAL tidb_enable_row_level_checksum = ON;
 ```
 
+Now you need to re-connect to activate this for your session.
```
Comment (Oreoxmt) [inline functions-and-operators/tidb-functions.md:336]:
```suggestion
This configuration only takes effect for newly created sessions, so you need to reconnect to TiDB.
```

**Review by Oreoxmt — APPROVED** [review]
Rest LGTM

**Pull request merged**